pkg/controller/direct/gkehub/featuremembership_controller.go

```diff
@@ -249,7 +248,7 @@ func (a *gkeHubAdapter) Update(ctx context.Context, updateOp *directbase.UpdateO
 	log.V(2).Info("updating object", "u", u)
 	actual := a.actual.MembershipSpecs[a.membershipID]
 	//  There are no output fields in the api Object, so we can compare the desired and the actaul directly.
-	if !reflect.DeepEqual(a.desired.Configmanagement, actual.Configmanagement) || !reflect.DeepEqual(a.desired.Policycontroller, actual.Policycontroller) || !reflect.DeepEqual(a.desired.Mesh, actual.Mesh) {
```

Comment:
What's the problem of DeepEqual here? I think we use it a lot in diffing for other resources. If there are any particular issues, we should identify them here and avoid them for future implementations.

Reply:
the `desired `is converted from KRM representation. The `actual` is actually fetched from the gcp server. If there are server defaults values, there will be permanent diff in the controller.